### PR TITLE
unit-tests: Increase the test coverage for unit tests

### PR DESCRIPTION
**Pull request by oleorhagen**
* Add unit tests for missing modules (not all)
* Functionality: Upload deployment logs upon errors

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -39,17 +40,21 @@ test:static:
     before_script:
         - pip install --cache-dir=~/.cache/pip-test mypy
         - pip install --cache-dir=~/.cache/pip-test pylint
+        - pip install --cache-dir=~/.cache/pip-test .
         - pip install --cache-dir=~/.cache/pip-test -r requirements.txt
+        - pip install --cache-dir=~/.cache/pip-test -r tests/unit/requirements.txt
     script:
         - mypy src
         - pylint --rcfile=.pylintrc-develop src
+        - pylint --rcfile=.pylintrc-develop --disable=no-self-use --disable=no-member --disable=too-many-ancestors tests/unit/test_*.py
```
Comment (lluiscampos) [inline .gitlab-ci.yml:49]: I like it.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -39,17 +40,21 @@ test:static:
...
 
 test:unit:
     extends: .test
     stage: test
     image: python:3.6-buster
     before_script:
         - pip install --cache-dir=~/.cache/pip-test/ -r requirements.txt
+        - pip install --cache-dir=~/.cache/pip-test -r tests/unit/requirements.txt
```
Comment (lluiscampos) [inline .gitlab-ci.yml:57]: nit pick for consistency
```suggestion
        - pip install --cache-dir=~/.cache/pip-test/ -r tests/unit/requirements.txt
```

**Review by lluiscampos — APPROVED** [review]
Other than this PR does more than the title says, I cannot spot any error.

**Comment by oleorhagen** [comment]
Fixed, merging.

**Pull request merged**